ignite/services/plugin/template/main.go.plush

```diff
@@ -2,117 +2,45 @@ package main
-	return &plugin.Manifest{
-		Name: "<%= Name %>",
-		// TODO: Add commands here
-		Commands: []*plugin.Command{
-			// Example of a command
-			{
-				Use:   "<%= Name %>",
-				Short: "Explain what the command is doing...",
-				Long:  "Long description goes here...",
-				Flags: []*plugin.Flag{
-					{Name: "my-flag", Type: plugin.FlagTypeString, Usage: "my flag description"},
-				},
-				PlaceCommandUnder: "ignite",
-				// Examples of adding subcommands:
-				// Commands: []*plugin.Command{
-				// 	{Use: "add"},
-				// 	{Use: "list"},
-				// 	{Use: "delete"},
-				// },
-			},
-		},
-		// TODO: Add hooks here
-		Hooks: []*plugin.Hook{},
-		SharedHost: <%= SharedHost %>,
-	}, nil
-}
-
-func (app) Execute(ctx context.Context, cmd *plugin.ExecutedCommand, api plugin.ClientAPI) error {
-	// TODO: write command execution here
-	fmt.Printf("Hello I'm the example-plugin plugin\n")
-	fmt.Printf("My executed command: %q\n", cmd.Path)
-	fmt.Printf("My args: %v\n", cmd.Args)
-
-	flags, err := cmd.NewFlags()
-	if err != nil {
-		return err
+func (app) Manifest(_ context.Context) (*plugin.Manifest, error) {
+	m := &plugin.Manifest{
+		Name: "<%= Name %>",<%= if (SharedHost) { %>
+		SharedHost: true,<% } %>
 	}
-
-	myFlag, _ := flags.GetString("my-flag")
-	fmt.Printf("My flags: my-flag=%q\n", myFlag)
-	fmt.Printf("My config parameters: %v\n", cmd.With)
-
-	// This is how the plugin can access the chain:
-	// c, err := getChain(cmd)
-	// if err != nil {
-	// 	return err
-	// }
-
-	// According to the number of declared commands, you may need a switch:
-	/*
-		switch cmd.Use {
-		case "add":
-			fmt.Println("Adding stuff...")
-		case "list":
-			fmt.Println("Listing stuff...")
-		case "delete":
-			fmt.Println("Deleting stuff...")
-		}
-	*/
-	
-	// ClientAPI call example
-	fmt.Println(api.GetChainInfo(ctx))
-	
-	return nil
+	m.ImportCobraCommand(cmd.New<%= Title %>(), "ignite")
+	return m, nil
 }
 
-func (app) ExecuteHookPre(ctx context.Context, h *plugin.ExecutedHook, api plugin.ClientAPI) error {
-	fmt.Printf("Executing hook pre %q\n", h.Hook.GetName())
-	return nil
+func (app) Execute(_ context.Context, c *plugin.ExecutedCommand, _ plugin.ClientAPI) error {
+	// Instead of a switch on c.Use, we run the root command like if
+	// we were in a command line context. This implies to set os.Args
+	// correctly.
+	// Remove the first arg "ignite" from OSArgs because our <%= Name %>
+	// command root is "<%= Name %>" not "ignite".
+	os.Args = c.OsArgs[1:]
+	return cmd.New<%= Title %>().Execute()
 }
 
-func (app) ExecuteHookPost(ctx context.Context, h *plugin.ExecutedHook, api plugin.ClientAPI) error {
-	fmt.Printf("Executing hook post %q\n", h.Hook.GetName())
+func (app) ExecuteHookPre(_ context.Context, _ *plugin.ExecutedHook, _ plugin.ClientAPI) error {
 	return nil
 }
 
-func (app) ExecuteHookCleanUp(ctx context.Context, h *plugin.ExecutedHook, api plugin.ClientAPI) error {
-	fmt.Printf("Executing hook cleanup %q\n", h.Hook.GetName())
```

Comment:
Why was this removed?

Reply:
@Pantani @julienrbrt  I already answered this question here https://github.com/ignite/cli/pull/3839#discussion_r1429755319
 Like i said the focus of this PR is to minimize the code we generate down to a minimal app so that developers won't need to remove any code. I think that these hook implementations won't be used by majority of the apps so there's no point in keeping them in the template.